clang/test/SemaSYCL/intel-reqd-work-group-size.cpp

```diff
@@ -0,0 +1,127 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -verify -DTRIGGER_ERROR %s
+// RUN: %clang_cc1 -fsycl-is-device -ast-dump %s | FileCheck %s
+// RUN: %clang_cc1 -fsycl-is-host -fsyntax-only -verify %s
+
+#ifndef __SYCL_DEVICE_ONLY__
+// expected-no-diagnostics
+class Functor {
+public:
+  [[intel::reqd_work_group_size(4, 1, 1)]] void operator()() {}
+};
+
+template <typename name, typename Func>
+void kernel(Func kernelFunc) {
+  kernelFunc();
+}
+
+void bar() {
+  Functor f;
+  kernel<class kernel_name>(f);
+}
+#else
+[[intel::reqd_work_group_size(4, 1, 1)]] void f4x1x1() {} // expected-note {{conflicting attribute is here}}
+// expected-note@-1 {{conflicting attribute is here}}
+[[intel::reqd_work_group_size(32, 1, 1)]] void f32x1x1() {} // expected-note {{conflicting attribute is here}}
+
+[[intel::reqd_work_group_size(16, 1, 1)]] void f16x1x1() {}   // expected-note {{conflicting attribute is here}}
```

Comment:
Since the defaults are 1, why set them so here (and around here)?  

Since the advantage to this form of attribute is the 'default' values, perhaps we should have more tests evaluating that.

Reply:
Okey.